src/csharp/Microsoft.Spark/Interop/Ipc/PayloadHelper.cs

```diff
@@ -343,35 +334,17 @@ internal static byte[] GetTypeId(Type type)
                         return s_arrayTypeId;
                     }
 
-                    if (IsDictionary(type))
-                    {
-                        return s_dictionaryTypeId;
-                    }
-
-                    if (typeof(IEnumerable<IJvmObjectReferenceProvider>).IsAssignableFrom(type))
-                    {
-                        return s_arrayTypeId;
-                    }
-
-                    if (typeof(IEnumerable<GenericRow>).IsAssignableFrom(type))
-                    {
-                        return s_rowArrTypeId;
-                    }
-
-                    if (typeof(IEnumerable<object>).IsAssignableFrom(type))
+                    return type switch
                     {
-                        return s_objectArrTypeId;
-                    }
-
-                    if (typeof(Date).IsAssignableFrom(type))
-                    {
-                        return s_dateTypeId;
-                    }
-
-                    if (typeof(Timestamp).IsAssignableFrom(type))
-                    {
-                        return s_timestampTypeId;
-                    }
+                        _ when typeof(IJvmObjectReferenceProvider).IsAssignableFrom(type) => s_jvmObjectTypeId,
```

Comment:
I think some of these lines are too long.  Can you check https://github.com/dotnet/spark/blob/main/docs/coding-guidelines/csharp-coding-style.md  ?

Reply:
it is exactly at the limit
![Screenshot 2021-12-08 at 01 23 42](https://user-images.githubusercontent.com/89642092/145126397-704c88e7-aae3-4a30-be92-b4e67052d655.png)